### PR TITLE
[docs-beta] Add dagster.yaml reference

### PR DESCRIPTION
**Pull request by PedramNavid**
## Summary & Motivation

I tried my best to figure out what the schema is of `dagster.yaml`. Mostly looked at the following resources:

https://docs.dagster.io/deployment/dagster-instance
https://docs.dagster.io/guides/limiting-concurrency-in-data-pipelines#limiting-opasset-concurrency-across-runs
https://github.com/dagster-io/dagster/blob/89431cdf20e557ed9d232ba7bd417c7cccb7fbc0/python_modules/dagster/dagster/_core/instance/config.py#L333

There were lots of inconsistencies here. For example, the table says the key is `data_retention` but the example shows `retention`. 
I wasn't clear on why `QueuedRunCoordinator` has concurreny limits in the config, but the concurrency guide seems to suggest there's another top-level config.
The source code helps a little, but many fields aren't documented.

Hoping someone more familiar with this can help validate this. In the future, we might consider documenting the config as a Pydantic class and auto-generating documentation.

## How I Tested These Changes

I didn't

## Changelog

Insert changelog entry or "NOCHANGELOG" here.

NOCHANGELOG

### CODE REVIEW

**Comment by PedramNavid** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24867?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24867** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24867?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24866** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24866?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24863** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24863?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @PedramNavid and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/docs-beta/docs/reference/dagster-yaml.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+  sqlite:
+    base_dir: /path/to/sqlite/storage
+  # Or use postgres:
+  # postgres:
+  #   postgres_db:
+  #     hostname: localhost
+  #     username: dagster
+  #     password: dagster
+  #     db_name: dagster
+  #     port: 5432
```
Comment (cmpadden) [inline docs/docs-beta/docs/reference/dagster-yaml.md:55]: Might be worth using environment variables in this example, as that is what people should end up doing.

File: docs/docs-beta/docs/reference/dagster-yaml.md [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+  module: dagster.core.storage.local_compute_log_manager
+  class: LocalComputeLogManager
+  config:
+    base_dir: /path/to/compute/logs
+  # Can use S3, GCS, or Azure Blob Storage for compute logs, e.g.
+  # module: dagster_aws.s3.compute_log_manager
+  # class: S3ComputeLogManager
+  # config:
+  #   bucket: "mycorp-dagster-compute-logs"
+  #   prefix: "dagster-test-"
```
Comment (cmpadden) [inline docs/docs-beta/docs/reference/dagster-yaml.md:43]: What if we separate the alternative into its own block, like so?
```suggestion
compute_logs:
  module: dagster.core.storage.local_compute_log_manager
  class: LocalComputeLogManager
  config:
    base_dir: /path/to/compute/logs
    
# Alternatively, logs can be written to cloud storage providers like S3, GCS, and Azure blog storage. For example:
# compute_logs:
#   module: dagster_aws.s3.compute_log_manager
#   class: S3ComputeLogManager
#   config:
#     bucket: "mycorp-dagster-compute-logs"
#    prefix: "dagster-test-"
```

**Review by cmpadden — COMMENTED** [review]
This is tremendously helpful!

Only left a small comment around how we present the alternative configurations in the comprehensive example as it was slightly overwhelming at first. But open to push back.

**Review by cmpadden — APPROVED** [review]

**Comment by PedramNavid** [comment]
### Merge activity

* **Oct 1, 7:17 PM EDT**: @PedramNavid merged this pull request with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/24867).

**Pull request merged**